File: auction-app-backend/src/main/java/com/internship/auctionapp/requests/SearchProductRequest.java
package com.internship.auctionapp.requests;

import com.internship.auctionapp.util.PageableRequest;
import com.internship.auctionapp.util.filter.product.ProductSort;
import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
public class SearchProductRequest extends PageableRequest<String> {
    private Integer pageNumber;
    private String name;
    private UUID categoryId;
    private List<UUID> subcategoryIds;
    private Double minPrice;
    private Double maxPrice;
    private ProductSort productSort;

    private final static Integer DEFAULT_PAGE_SIZE = 9;

    public SearchProductRequest(Integer pageNumber) {
        super(pageNumber, DEFAULT_PAGE_SIZE, null);
    }
}

File: auction-app-ui/src/models/request/search/searchProductRequest.ts
import { SortingOption } from 'models/enum/sortingOption';

export type SearchProductRequest = {
  pageNumber: number;
  name?: string;
  categoryId?: string;
  subcategoryIds?: string[];
  minPrice?: number;
  maxPrice?: number;
  productSort?: SortingOption;
};
